doc/source/tutorial/stats/sampling_pinv.rst

```diff
@@ -0,0 +1,271 @@
+.. _sampling-pinv:
+
+Polynomial interpolation based INVersion of CDF (PINV)
+======================================================
+
+.. currentmodule:: scipy.stats
+
+* Required: PDF
+* Optional: CDF, mode, center
+* Speed:
+
+  * Set-up: (very) slow
+  * Sampling: (very) fast
+
+Polynomial interpolation based INVersion of CDF (PINV) is an inversion method
+that only requires the density function to sample from a distribution. It is
+based on Polynomial interpolation of the PPF and Gauss-Lobatto integration
+of the PDF. It provides control over the interpolation error and integration
+error. Its primary purpose is to provide very fast sampling which is nearly
+the same for any given distribution at the cost of moderate to slow setup
+time. It is the fastest known inversion method for the fixed-parameter case.
+
+The inversion method is the simplest and most flexible to sample nonuniform
+random variates. For a target distribution with CDF :math:`F` and a uniform
+random variate :math:`U` sampled from :math:`\text{Uniform}(0, 1)`, a random
+variate X is generated by transforming the uniform random variate :math:`U`
+using the PPF (inverse CDF) of the distribution:
+
+.. math::
+    X = F^{-1}(U)
+
+This method is suitable for stochastic simulations because of its advantages.
+Some of the most attractive are:
+
+* It preserves the structural properties of the uniform random number sampler.
+* Transforms a uniform random variate :math:`U` one-to-one into non-uniform
+  random variates :math:`X`.
+* Easy and efficient sampling from truncated distributions.
+
+Unfortunately, the PPF is rarely available in closed form or too slow when
+available. For many distributions, the CDF is also not easy to obtain. This
+method addresses both the shortcomings. The user only has to provide the PDF
+and optionally a point near the mode (called "center") together with the size
+of the maximal acceptable error. It uses a combination of an adaptive and a
+simple Gauss-Lobatto quadrature to obtain the CDF and Newton's interpolation
+to obtain the PPF. The method is not exact, as it only produces random
+variates of the approximated distribution. Nevertheless, the maximal tolerated
+approximation error can be set close to the machine precision and its relative
```

Comment:
what does relative error refer to?

Reply:
I think the relative x-error:

![image](https://user-images.githubusercontent.com/43181252/130560357-21080b95-2f97-4950-85e1-00b448cc0b1d.png)

Reply:
I am still confused about this: we only control the u-error, not the x-error. Is the rel x-error really below 1e-15?

Reply:
> I am still confused about this: we only control the u-error, not the x-error. Is the rel x-error really below 1e-15?

The paper mentions it. Though, I didn't see any proof. This might be a heuristic based on their experiments.

Reply:
Hm, I did not see it right away in the paper. Could you point me to it?

Reply:
I think the two highlighted sentences say that the relative x-error achieved by the algorithm is near to machine precision (1e-15):

![image](https://user-images.githubusercontent.com/43181252/130669426-6c27fbcd-4dfc-4128-9c89-c6e2eb1b7b51.png)

Though, the second highlighted sentence might be talking about the algorithm that uses the x-error as the criterion. What do you think?

Reply:
i think the authors mean the following:

- if the u-error is ~1e-15, they will call the algorithm exact
- then they explain that the absolute x-error has some limitations (approximation needs to be very accurate in the tails)
- if the relative x-error is used, you avoid the problem in the fails but you get a problem at 0

They therefore conclude that the u-error is the right measure (only the absolute one is considered since u is in (0,1) in the expression |u -F(...)|)

and the u-error and x-error are related:
![image](https://user-images.githubusercontent.com/33071866/130781588-4d91c380-9ce4-4312-90e3-415a6de542e7.png)

but in general, you cannot guarantee that the x-error is small if the u-error is

if we just say "Nevertheless, the maximal tolerated approximation error can be set close to the machine precision", it should be fine.

Reply:
I think I understand it now. Thanks for clearing that out!

> if we just say "Nevertheless, the maximal tolerated approximation error can be set close to the machine precision", it should be fine.

Agreed. I will change it this way.

Reply:
Changed now. Thanks, @chrisb83!